src/widgets/barchart/mod.rs

```diff
@@ -122,12 +120,39 @@ impl<'a> BarChart<'a> {
         self
     }
 
+    pub fn group_gap(mut self, gap: u16) -> BarChart<'a> {
+        self.group_gap = gap;
+        self
+    }
+
     pub fn style(mut self, style: Style) -> BarChart<'a> {
         self.style = style;
         self
     }
 }
 
+impl<'a> BarChart<'a> {
+    fn remove_unvisible_groups_and_bars(&mut self, mut width: u16) {
```

Comment:
This method doesn't have an obvious reason to exist (which makes it difficult to judge whether it works correctly and difficult to maintain). Can you please write a doc comment that helps clarify why it exists and what it should do.